README.md

```diff
@@ -45,6 +45,178 @@ To run tests against a specific host, specify in the environment:
 
 Tests will run against sandbox by default.
 
+## Installation ##
+
+Download [the latest JAR](https://github.com/ably/ably-java/releases) or grab via Gradle:
+
+```groovy
+compile 'io.ably:ably-java:0.8.1'
+```
+
+and add following repo for a sub-dependency,
+
+```groovy
+maven { url "https://raw.github.com/paddybyers/Java-WebSocket/mvn-repo/" }
+```
+
+## Using the Realtime API ##
+
+### Introduction ###
+
+All examples assume a client has been created as follows:
+
+```java
+AblyRealtime ably = new AblyRealtime("xxx");
+```
+
+### Connection ###
+
+AblyRealtime will attempt to connect automatically once new instance is created. Also, it offers API for listening connection state changes.
+
+```java
+ably.connection.on(new ConnectionStateListener() {
+	@Override
+	public void onConnectionStateChanged(ConnectionStateChange state) {
+		switch (state.current) {
+			case connected: {
+				// Successful connection
+				break;
+			}
+			case failed: {
+				// Failed connection
+				break;
+			}
+		}
+	}
+});
+```
+
+### Subscribing to a channel ###
+
+Given:
+
+```java
+Channel channel = ably.channels.get("test");
+```
+
+Subscribe to all events:
+
+```java
+channel.subscribe(new MessageListener() {
+	@Override
+	public void onMessage(Message[] messages) {
+	}
+});
+```
+
+or subscribe to certain events:
+
+```java
+String[] events = new String[] {"event1", "event2"};
+channel.subscribe(events, new MessageListener() {
+	@Override
+	public void onMessage(Message[] messages) {
+	}
+});
+```
+
+### Publishing to a channel ###
+
+```java
+channel.publish("greeting", "Hello World!", new CompletionListener() {
+	@Override
+	public void onSuccess() {
+		// Successfully published message
+	}
+
+	@Override
+	public void onError(ErrorInfo reason) {
+		// Failed to publish message
+	}
+});
+```
+
+### Querying the history ###
+
+```java
+PaginatedResult<Message> result = channel.history(null);
```

Comment:
I concur. I think it would be useful to show the objects that are returned and typical values, see https://github.com/ably/ably-ruby for an example. Equally, you could certainly link to the docs for more detail on the types, perhaps link to docs.ably.io for now, and we can update the links to www.ably.io/documentation later.  Is that OK @gokhanbarisaker?